### PR TITLE
Unit tests for the createWithSigningFunction

### PR DESCRIPTION
**Pull request by anthonyvercolano**
# Description of the problem
Needed unit tests for createWithSigningFunction

# Description of the solution
Wrote some unit tests.

### CODE REVIEW

File: common/core/devdoc/shared_access_signature.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+### toString()
+Creates a string representation of the SharedAccessSignature.
+
+SRS_NODE_COMMON_SAS_05_019: [The toString method shall return a shared-access signature token of the form: SharedAccessSignature sr=<resourceUri>&sig=<urlEncodedSignature>&se=<expiry>&skn=<urlEncodedKeyName>]
+SRS_NODE_COMMON_SAS_05_020: [The skn segment is not part of the returned string if the skn property is not defined.]
+
+### createWithSigningFunction(credentials: authorization.TransportConfig, expiry: number, signingFunction: Function, callback: (err: Error, sas?: SharedAccessSignature) => void) [static]
+Creates a `SharedAccessSignature` utilizing the supplied `signingFunction`.  If the signing operation is successful the results will be made available via the `callback`
+
+**SRS_NODE_COMMON_SAS_06_001: [** If `credentials`, `expiry`, `signingFunction`, or `callback` are falsy, `createWithSigningFunction` shall throw ReferenceException. **]**
```
Comment (pierreca) [inline common/core/devdoc/shared_access_signature.md]: nit: ReferenceError. ReferenceException does not exist

File: common/core/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "alltest": "istanbul cover node_modules/mocha/bin/_mocha -- --reporter spec test/_*_test*.js",
     "ci": "npm -s run lint && npm -s run build && npm -s run alltest-min && npm -s run check-cover",
     "test": "npm -s run lint && npm -s run build && npm -s run unittest",
-    "check-cover": "istanbul check-coverage --statements 95 --branches 84 --functions 95 --lines 95"
+    "check-cover": "istanbul check-coverage --statements 99 --branches 97 --functions 98 --lines 99"
```
Comment (pierreca) [inline common/core/package.json]: nice!

File: common/core/test/_shared_access_signature_test.js [hunk 3]
```diff
@@ -175,6 +176,121 @@ describe('SharedAccessSignature', function () {
...
+      };
+      var fakeSignedValue = Buffer.from('abcd');
+      var fakeSigningFunction = sinon.stub().callsArgWith(1, null, fakeSignedValue);
+      SharedAccessSignature.createWithSigningFunction(fakeCredentials, 1, fakeSigningFunction, (err, newSas) => {
+        assert.strictEqual(newSas.se, 1, 'Invalid expiry create for sas token');
+        callback();
+      });
+    });
+
+    it('creates a valid skn value', function(callback) {
```
Comment (pierreca) [inline common/core/test/_shared_access_signature_test.js]: not requirement for this one?

File: common/core/package.json (hunk 2 above)
Comment (anthonyvercolano) [reply common/core/package.json]: Yes.

---
In reply to: [165243974](https://github.com/Azure/azure-iot-sdk-node/pull/217#discussion_r165243974) [](ancestors = 165243974)

File: common/core/devdoc/shared_access_signature.md (hunk 1 above)
Comment (anthonyvercolano) [reply common/core/devdoc/shared_access_signature.md]: Derp.

---
In reply to: [165243853](https://github.com/Azure/azure-iot-sdk-node/pull/217#discussion_r165243853) [](ancestors = 165243853)

File: common/core/test/_shared_access_signature_test.js (hunk 3 above)
Comment (anthonyvercolano) [reply common/core/test/_shared_access_signature_test.js]: No, I just forgot to copy it.

---
In reply to: [165244629](https://github.com/Azure/azure-iot-sdk-node/pull/217#discussion_r165244629) [](ancestors = 165244629)

**Pull request merged**